app/models/subscription/incomesource/SignUpRequest.scala

```diff
@@ -65,8 +65,17 @@ object BusinessIncomeModel {
 object PropertyIncomeModel {
   implicit val format: OFormat[PropertyIncomeModel] = Json.format[PropertyIncomeModel]
 
-  def writeToDes(propertyIncomeModel: PropertyIncomeModel) = {
-    propertyIncomeModel.accountingMethod.fold(Json.obj())(accountingMethod => Json.obj("cashOrAccruals" -> accountingMethod))
+  def writeToDes(propertyIncomeModel: PropertyIncomeModel): JsObject = {
+
+    def toDesCashAccruals: AccountingMethod => String = {
```

Comment:
This would result in a new function being generated every time this def is called which would be inefficient. When I suggested removing the function literal, I meant doing `def toDesCashAccruals(accountingMethod: AccountingMethod): String = {`

Although you could simplify the whole thing further by just doing:
```
  val cashAccrualsFlag = "cashAccrualsFlag"

  def writeToDes(propertyIncomeModel: PropertyIncomeModel): JsObject = {
    propertyIncomeModel.accountingMethod match {
      case Some(Cash) => Json.obj(cashAccrualsFlag -> "C")
      case Some(Accruals) => Json.obj(cashAccrualsFlag -> "A")
      case None => Json.obj()
    }
  }
```
And removing the entire `toDesCashAccruals` method.